frontend/src/app.js

```diff
@@ -22,16 +21,24 @@ import './algolia.css';
 import './app.css';
 import './modal';
 import { sounds } from './player';
+import { restoreSearchWidgets } from './dom-utils';
+import {AlgoliaWidgets} from './algolia-widgets';
```

Comment:
It is very good that you are introducing some decoupling of the code. We will definitely have to split the code into small modules. However we will do that later as a part of a separate task/PR. This task does not necessarily require this step, therefore please reduce the changes to the minimum code that is required to achieve the functionality.

See 5584aa153412b687c3c37b8477be230f42f95a88 for example of adding a loading indicator. Very small changes, related only to the task itself. Also see [our guidelines](https://github.com/PureBhaktiArchive/archive-web-app/blob/main/CONTRIBUTING.md#:~:text=Don't%20include%20irrelevant%20changes%20into%20commits).